File: leetCode/groupAnagrams/java/GroupAnagrams.java
public class GroupAnagrams {

    public static void main(String[] args) {
        List<String> input = List.of("eat", "tea", "tan", "ate", "nat", "bat");

        Map<String, List<String>> anagramMap = new HashMap<>();

        for (String s : input) {
            String sortedString = sortString(s);
            if (anagramMap.containsKey(sortedString)) {
                anagramMap.get(sortedString).add(s);
            } else {
                anagramMap.put(sortedString, new ArrayList<>(List.of(s)));
            }
        }

        System.out.println(anagramMap.values());
    }

    private static String sortString(String s) {
        char[] stringChars = s.toCharArray();
        Arrays.sort(stringChars);
        return String.valueOf(stringChars);
    }
}

File: leetCode/groupAnagrams/rust/group_anagrams.rs
use std::collections::HashMap;

fn main() {
    group_anagrams(vec!["eat".to_string(), "tea".to_string(), "tan".to_string(), "ate".to_string(), "nat".to_string(), "bat".to_string()]);
}

// Working but need more refinement and understanding
fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
    let mut h = HashMap::new();

    for s in strs {
        let mut key: Vec<char> = s.chars().collect();
        key.sort();
        h.entry(key).or_insert(vec![]).push(s);
    }

    println!("{:?}", h.values().cloned().collect::<Vec<_>>());

    return vec![vec!["xd".to_string()]];
}